raft/src/main/java/org/apache/kafka/raft/utils/DescribeQuorumRpc.java

```diff
@@ -0,0 +1,135 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.raft.utils;
+
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.message.DescribeQuorumRequestData;
+import org.apache.kafka.common.message.DescribeQuorumResponseData;
+import org.apache.kafka.common.protocol.Errors;
+import org.apache.kafka.raft.LeaderState;
+import org.apache.kafka.raft.LogOffsetMetadata;
+import org.apache.kafka.raft.ReplicaKey;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DescribeQuorumRpc {
+    public static DescribeQuorumRequestData singletonDescribeQuorumRequest(
+        TopicPartition topicPartition
+    ) {
+        return new DescribeQuorumRequestData()
+            .setTopics(
+                Collections.singletonList(
+                    new DescribeQuorumRequestData.TopicData()
+                        .setTopicName(topicPartition.topic())
+                        .setPartitions(
+                            Collections.singletonList(
+                                new DescribeQuorumRequestData.PartitionData()
+                                    .setPartitionIndex(topicPartition.partition())
+                            )
+                        )
+                )
+            );
+    }
+
+    public static DescribeQuorumResponseData singletonDescribeQuorumResponse(
+        short apiVersion,
+        TopicPartition topicPartition,
+        int leaderId,
+        int leaderEpoch,
+        long highWatermark,
+        Collection<LeaderState.ReplicaState> voters,
+        Collection<LeaderState.ReplicaState> observers,
+        long currentTimeMs
+    ) {
+        DescribeQuorumResponseData response = new DescribeQuorumResponseData()
+            .setTopics(
+                Collections.singletonList(
+                    new DescribeQuorumResponseData.TopicData()
+                        .setTopicName(topicPartition.topic())
+                        .setPartitions(
+                            Collections.singletonList(
+                                new DescribeQuorumResponseData.PartitionData()
+                                    .setPartitionIndex(topicPartition.partition())
+                                    .setErrorCode(Errors.NONE.code())
+                                    .setLeaderId(leaderId)
+                                    .setLeaderEpoch(leaderEpoch)
+                                    .setHighWatermark(highWatermark)
+                                    .setCurrentVoters(toReplicaStates(apiVersion, leaderId, voters, currentTimeMs))
+                                    .setObservers(toReplicaStates(apiVersion, leaderId, observers, currentTimeMs))))));
```

Comment:
style nit: I know this was carried over from the old code, but maybe better to have each closing parenthesis on its own new line

Reply:
I will take a look 😺